tests/performance/PerformanceTest.cpp

```diff
@@ -0,0 +1,179 @@
+        .WithUsername("test_user")
+        .WithPassword("test_password")
+        .WithType(olp::http::NetworkProxySettings::Type::HTTP);
+  }
+
+  std::shared_ptr<dataservice::read::CatalogClient> CreateCatalogClient() {
+    client::HRN hrn("fake");
+
+    auto parameter = GetParam();
+
+    std::shared_ptr<olp::thread::TaskScheduler> task_scheduler =
+        olp::client::OlpClientSettingsFactory::CreateDefaultTaskScheduler(
+            parameter.task_scheduler_capacity);
+
+    olp::authentication::Settings settings;
+    settings.task_scheduler = task_scheduler;
+    settings.network_request_handler = s_network;
+
+    olp::client::AuthenticationSettings auth_settings;
+    auth_settings.provider = olp::authentication::TokenProviderDefault(
+        kKeyId, kKeySecret, std::move(settings));
+
+    auto client_settings = std::make_shared<olp::client::OlpClientSettings>();
+    client_settings->authentication_settings = auth_settings;
+    client_settings->task_scheduler = task_scheduler;
+    client_settings->network_request_handler = s_network;
+    client_settings->proxy_settings = GetLocalhostProxySettings();
+
+    std::shared_ptr<olp::cache::KeyValueCache> cache = CreateCache();
+
+    return std::make_shared<dataservice::read::CatalogClient>(
+        hrn, client_settings, cache);
+  }
+
+  std::shared_ptr<olp::cache::KeyValueCache> CreateCache() {
+    using ::testing::_;
+    using ::testing::Return;
+    if (GetParam().use_no_cache) {
+      auto cache = std::make_shared<olp::testing::mocks::KeyValueCache>();
+      ON_CALL(*cache, Put(_, _, _, _)).WillByDefault(Return(true));
+      ON_CALL(*cache, Put(_, _, _)).WillByDefault(Return(true));
+      ON_CALL(*cache, Get(_, _)).WillByDefault(Return(boost::any()));
+      ON_CALL(*cache, Get(_, _)).WillByDefault(Return(nullptr));
+      ON_CALL(*cache, Remove(_)).WillByDefault(Return(true));
+      ON_CALL(*cache, RemoveKeysWithPrefix(_)).WillByDefault(Return(true));
+      return cache;
+    } else {
+      return olp::dataservice::read::CreateDefaultCache();
+    }
+  }
+
+ protected:
+  static std::shared_ptr<olp::http::Network> s_network;
+};
+
+std::shared_ptr<olp::http::Network> CatalogClientTest::s_network;
+
+TEST_P(CatalogClientTest, ReadNPartitions) {
+  auto parameter = GetParam();
+
+  auto service_client = CreateCatalogClient();
+
+  std::vector<std::thread> client_threads;
+
+  for (int client_id = 0; client_id < parameter.calling_thread_count;
+       client_id++) {
+    auto thread = std::thread([&]() {
+      for (int iteration = 0; iteration < parameter.number_of_requests;
+           iteration++) {
+        auto request = olp::dataservice::read::DataRequest()
+                           .WithLayerId(kLayerId)
+                           .WithPartitionId(kPartitionId)
+                           .WithBillingTag(boost::none);
+
+        auto future = service_client->GetData(request);
+        olp::dataservice::read::DataResponse data_response =
```

Comment:
what happens when 5 request succeed and 995 return kSlowDown?
Is this still a real performance test?
Please check error and react accordingly.